commands/add.js

```diff
@@ -109,14 +166,14 @@ function prepUploadOperation(message, list, row) {
 				return;
 			}
 			imageLocation = resp.groups.link;
-		} else if (row.nh.match(/imgur/) !== null) {
-			let hashCode = /https:\/\/imgur.com\/a\/([A-z0-9]*)/.exec(row.nh)[1];
+		} else if (row?.im?.match(/imgur/)) {
```

Comment:
I would move the Imgur check above nh, since if an nh link exists, we can't scrape the thumbnail anyway

In that case, we should prefer imgur if it exists